packages/features/timezone-buddy/components/TimeDial.tsx

```diff
@@ -68,57 +102,114 @@ export function TimeDial({ timezone, dateRanges }: TimeDialProps) {
-                    )}
-                    style={{
-                      width: `${DAY_CELL_WIDTH}px`,
-                    }}>
-                    {hours ? (
-                      <div title={hourSet.format("DD/MM HH:mm")}>
+            <div key={i} className={classNames("border-subtle overflow-hidden rounded-lg border-2")}>
+              <div className="flex flex-none">
+                {day.map((h) => {
+                  const hours = Math.floor(h); // Whole number part
+                  const fractionalHours = h - hours; // Decimal part
+
+                  // Convert the fractional hours to minutes
+                  const minutes = fractionalHours * 60;
+                  const hourSet = dateWithDaySet.set("hour", h).set("minute", minutes);
+
+                  const { isInRange, rangeOverlap = 0 } = isCurrentHourInRange({
+                    dateRanges,
+                    offset,
+                    cellDate: hourSet,
+                  });
+
+                  const rangeGradients: {
+                    backgroundGradient?: string;
+                    textGradient?: string;
+                    darkTextGradient?: string;
+                  } = {};
+
+                  if (isInRange && rangeOverlap) {
+                    if (rangeOverlap < 0) {
+                      const gradientValue = Math.abs(rangeOverlap);
+                      rangeGradients.backgroundGradient = `linear-gradient(90deg, rgba(0,0,0,0) 0%, rgba(0,0,0,0) ${gradientValue}%, var(--cal-bg-success) ${gradientValue}%)`;
+
+                      rangeGradients.textGradient = `linear-gradient(90deg, rgba(0,212,255,1) 0%, rgba(2,0,36,1) 100%, rgba(9,108,121,1) 100%)`;
+
+                      rangeGradients.darkTextGradient = `linear-gradient(90deg, var(--cal-text-emphasis, #111827) ${
+                        gradientValue === 50 ? "50%" : Math.round(gradientValue / 100) * 100 + "%"
+                      }, var(--cal-text-inverted, white) 0%, var(--cal-text-inverted, white) 0%)`;
+                    } else {
+                      rangeGradients.backgroundGradient = `linear-gradient(90deg, var(--cal-bg-success) ${rangeOverlap}%, rgba(0,0,0,0) 0%, rgba(0,0,0,0) 0%)`;
+
+                      rangeGradients.textGradient = `linear-gradient(90deg, rgba(0,212,255,1) 0%, rgba(2,0,36,1) 50%, rgba(9,108,121,1) 100%)`;
+
+                      rangeGradients.darkTextGradient = `linear-gradient(90deg, var(--cal-text-inverted, white) ${
+                        rangeOverlap === 50 ? "50%" : Math.round(rangeOverlap / 100) * 100 + "%"
+                      }, var(--cal-text-emphasis, #111827) 0%, var(--cal-text-emphasis, #111827) 0%)`;
+                    }
+                  }
+
+                  const minuteOffsetStyles: { marginLeft?: string } = {};
+                  if (hours !== 0 && !minuteOffsetApplied) {
+                    minuteOffsetApplied = true;
+                    minuteOffsetStyles.marginLeft = `${DAY_CELL_WIDTH * (offset % 1)}px`;
```

Comment:
@emrysal X%1 is always 0 no?

Reply:
yes - that's intentional it should only shift for decimal places like 3.5 and 4.75